structure_tests/drivers/driver.go

```diff
@@ -37,14 +36,20 @@ type Driver interface {
 	ReadFile(t *testing.T, path string) ([]byte, error)
 
 	ReadDir(t *testing.T, path string) ([]os.FileInfo, error)
+
+	Destroy()
 }
 
-func InitDriverImpl(driver string) (func(string) Driver, error) {
+func InitDriverImpl(driver string) func(string) (Driver, error) {
 	switch driver {
 	// future drivers will be added here
 	case "docker":
-		return NewDockerDriver, nil
+		return NewDockerDriver
+	case "tar":
+		return NewTarDriver
 	default:
-		return nil, errors.New(fmt.Sprintf("Unsupported driver type: %s", driver))
+		return func(s string) (Driver, error) {
```

Comment:
Do you need this func, or can you just return nil, error

Reply:
need the func to stay consistent with this pattern, because this expects the return value to be `func(string) (Driver, error)` not `(Driver, error)`